compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -675,6 +675,26 @@ static TR::Register *shiftHelper(TR::Node *node, TR::ARM64ShiftCode shiftType, T
 TR::Register *
 OMR::ARM64::TreeEvaluator::ishlEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
+   if (node->getOpCodeValue() == TR::lshl)
+      {
+      TR::Node *firstChild = node->getFirstChild();
+      TR::Node *secondChild = node->getSecondChild();
+      if (firstChild->getOpCodeValue() == TR::i2l && secondChild->getOpCodeValue() == TR::iconst)
+         {
+         // Typical IL sequence in array access (sign-extend index value and shift it to left)
+         TR::Node *indexChild = firstChild->getFirstChild();
+         TR::Register *srcReg = cg->evaluate(indexChild);
+         TR::Register *trgReg = cg->allocateRegister();
+         int32_t bitsToShift = secondChild->getInt();
+         int32_t imm = ((64 - bitsToShift) << 6) | 31; // immr = 64 - bitsToShift, imms = 31
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::sbfmx, node, trgReg, srcReg, imm);
+         node->setRegister(trgReg);
+         cg->decReferenceCount(firstChild);
+         cg->decReferenceCount(secondChild);
```

Comment:
decReferenceCount on `indexChild` as well?

Reply:
I changed the code to call `recursivelyDecReferenceCount(firstChild)`.